<?xml version="1.0"?>
<launch>
  <!-- arguments -->
  <arg name="affordance_model_path" default="data/trained_res18.trt"/>
  <arg name="source_topic" default="/camera/rgb/image_rect_color"/>
  <arg name="camera_topic" default="/camera/rgb/camera_info"/>
  <arg name="depth_topic" default="/camera/depth_registered/hw_registered/image_rect_raw"/>
  <arg name="depth_camera_topic" default="/camera/depth_registered/camera_info"/>
  <arg name="detection_topic" default="/yolov5/detection"/>
  
  <arg name="save_img" default="False"/>  
  <arg name="publish" default="True"/>
  
  
  <!-- YOLOv5 node -->
  <node pkg="affordance_ros" type="trt_inference.py" name="affordance_node" output="screen">
    <param name="affordance_model_path" value="$(arg affordance_model_path)" />
    <param name="source_topic" value="$(arg source_topic)" />
    <param name="camera_topic" value="$(arg camera_topic)" />
    <param name="depth_topic" value="$(arg depth_topic)" />
    <param name="depth_camera_topic" value="$(arg depth_camera_topic)" />
    <param name="detection_topic" value="$(arg detection_topic)" />
    
    <param name="save_img" value="$(arg save_img)" />
    <param name="publish" value="$(arg publish)" />
    
  </node>

  
</launch>
